### ISSUE TITLE
Image export gives incorrect results when specifying size.

### ISSUE THREAD

**Issue by campagnola** [pyqtgraph/pyqtgraph#16]
When exporting with ImageExporter, specifying any width other than the default produces erroneous results. In some cases, axis labels are out of place, and in other cases lines do not match up to scatter plot points.

See this discussion: https://groups.google.com/d/msg/pyqtgraph/qkLUEiFt0Qk/pwiNToIKcMsJ

Example:

```
from PyQt4 import QtCore, QtGui
import pyqtgraph as pg
pg.mkQApp()

view = pg.PlotWidget(title='I0-Normalised reflectivity')
view.setLabel('left', "Reflectivity", units='arb. units')
view.setLabel('bottom', "Photon energy", units='eV')
view.showAxis('top', show=True)

exporter = pg.exporters.ImageExporter.ImageExporter(view.plotItem)
exporter.params['width'] = 1000
exporter.export('test.png')
```

**Comment by mfitzp** [comment]
Also experiencing this bug with scatterplots + plots with markers on lines. In both cases the markers end up taking up 50% of the area they should, being placed in the top left of the plot.

**Comment by aeroaks** [comment]
I tried changing the Widget for export and making it visible for processing the resizeevents and then again making it invisible. I get proper exported images using the following code, I do not change the Width parameter in the exporter class:

```
# plot export area
self.export_area = pg.GraphicsLayoutWidget(border=(50, 50, 50))
# Configure the colors:
self.export_area.setBackground('w')
# We resize the widget to our desired export dimensions.
# When the widget is hidden, the child widgets do not get resized.
# Only when the widget is made visible, the events are propagated to the child widgets.
# We immediately hide the widget again to prevent a pop-up or flashing screen.
self.export_area.resize(2000, 1000)
self.export_area.setVisible(True)
self.export_area.setVisible(False)
```

Then use self.export_area for drawing and exporting as an invisible widget

**Comment by 2xB** [comment]
Here is an updated version of the code given to reproduce this error with the latest pyqtgraph version from `develop`:

```python3
import pyqtgraph as pg
import pyqtgraph.exporters as ex
pg.mkQApp()

view = pg.PlotWidget(title='I0-Normalised reflectivity')
view.setLabel('left', "Reflectivity", units='arb. units')
view.setLabel('bottom', "Photon energy", units='eV')
view.showAxis('top', show=True)

exporter = ex.ImageExporter(view.plotItem)
exporter.params['width'] = 1000
exporter.export('test.png')
```

**Comment by j9ac9k** [comment]
Fixed with #88 being merged
